src/main/scala/com/fulcrumgenomics/vcf/AssessPhasing.scala

```diff
@@ -0,0 +1,780 @@
+      phase.zip(states.map(_ == 0)).count { case (p, s) => p != s }
+    }
+
+    IlluminaSwitchErrors(numPointErrors, numLongSwitchErrors(states), states.length)
+  }
+
+  /** Partitions the cigar into multiple cigars, with each cigar being a contiguous block of phased variants. */
+  private[vcf] def toPhasedBlocks(cigar: Cigar, isTruth: Boolean = false): Seq[Cigar] = {
+    val endCigarTypes = if (isTruth) Set(TruthEnd, BothEnd) else Set(CallEnd, BothEnd)
+
+    // Splits the cigar any time we reach a CigarType in the endCigarTypes
+    val (cigarsToReturn, lastCigar) = cigar.foldLeft((ListBuffer[Cigar](), ListBuffer[CigarType]())) {
+      case ((cigars: ListBuffer[Cigar], currentCigar: ListBuffer[CigarType]), cigarType: CigarType) =>
+        cigarType match {
+          case tpe if endCigarTypes.contains(tpe) => // split!
+            if (currentCigar.nonEmpty) cigars.append(currentCigar.toList)
+            (cigars, new ListBuffer[CigarType]())
+          case _ => // keep going
+            currentCigar.append(cigarType)
+            (cigars, currentCigar)
+        }
+    }
+    // Make sure to ge the last one
+    if (lastCigar.nonEmpty) cigarsToReturn.append(lastCigar)
+
+    cigarsToReturn.toList
+  }
+
+  /** Returns the cigar for the two variant contexts. */
+  private[vcf] def contextsToCigar(t: Option[VariantContext], c: Option[VariantContext]): Cigar = (t, c) match {
+    case (None, Some(c: VCtx))          if isStartOfPhaseBlock(c)                             => Seq(CallEnd, Call)
+    case (None, Some(c: VCtx))                                                                => Seq(Call)
+    case (Some(t: VCtx), None)          if isStartOfPhaseBlock(t)                             => Seq(TruthEnd, Truth)
+    case (Some(t: VCtx), None)                                                                => Seq(Truth)
+    case (Some(t: VCtx), Some(c: VCtx)) if isStartOfPhaseBlock(t)  && isStartOfPhaseBlock(c)  => Seq(BothEnd, cigarTypeForVariantContexts(t, c))
+    case (Some(t: VCtx), Some(c: VCtx)) if !isStartOfPhaseBlock(t) && isStartOfPhaseBlock(c)  => Seq(CallEnd, cigarTypeForVariantContexts(t, c))
+    case (Some(t: VCtx), Some(c: VCtx)) if isStartOfPhaseBlock(t)  && !isStartOfPhaseBlock(c) => Seq(TruthEnd, cigarTypeForVariantContexts(t, c))
+    case (Some(t: VCtx), Some(c: VCtx)) if !isStartOfPhaseBlock(t) && !isStartOfPhaseBlock(c) => Seq(cigarTypeForVariantContexts(t, c))
+    case _ => unreachable()
+  }
+
+  /** True if the phasing set id is the same as the start position of the given variant, false otherwise. */
+  def isStartOfPhaseBlock(ctx: VariantContext): Boolean = ctx.getStart == PhasingUtils.getPhasingSetId(ctx)
+
+  /** Computes the cigar for two variant contexts.  Returns [[Match]] if they share the same alleles in the same order,
+    * [[Mismatch]] otherwise.
+    */
+  private[vcf] def cigarTypeForVariantContexts(truth: VariantContext, call: VariantContext): CigarType = {
```

Comment:
This is a detail neither very well encoded in the VCF spec nor in the VCF code in `htsjdk`, but the order of alleles defines the phase!